pkglistgen/tool.py

```diff
@@ -763,7 +862,23 @@ def update_and_solve_target(
                 for line in sorted(output):
                     f.write(line + '\n')
 
-        self.commit_package(product_dir)
+        if git_url:
+            if os.path.isfile(f"{productcompose_dir}/default.productcompose"):
+                logging.debug(subprocess.check_output(
+                    ['git', 'add', productcompose_dir, f"{productcompose_dir}/default.productcompose"],
```

Comment:
`git add productcompose_dir` implies all files below, so `f"{productcompose_dir}/default.productcompose"` is redundant. Intentional?

Reply:
explanation was below:

"git add $DIR" implies all below files only when the directory is not yet added.

This way it works in both cases.

Reply:
cleaned up